src/pipeline_v3/cli/utils/validation.py

```
@@ -127,19 +161,25 @@ def validate_filter_expression(filter_str: str) -> dict[str, Any]:
-                raise ValidationError(f"Pattern too long (max 100 chars): {pattern}")
-
-            validated_patterns.append(pattern)
+        for pattern in patterns:
+            try:
+                # Use security validator for comprehensive checks
+                validated_pattern = PathSecurityValidator.validate_glob_pattern(
+                    pattern, allowed_dirs
+                )
+                validated_patterns.append(validated_pattern)
+            except SecurityError as e:
+                raise ValidationError(f"Security validation failed: {e}")
```

Pylint (3.3.7)

[warning] Consider explicitly re-raising using 'raise ValidationError(f'Security validation failed: {e}') from e'

(W0707)